src/components/FlightCard/FlightCard.tsx

```diff
@@ -1,23 +1,51 @@
+import React from "react";
 import { FlightResultsProps,  } from "../../pages/HomePage/interfaces";
-import { Button, Grid } from "@mui/material";
+import {Button, Grid, Typography} from "@mui/material";
 import flightCardStyles from "./FlightCard.module.css"
 import FlightInfoComponent from "./FlightInfoComponent";
+import FlightDetailsModal from "../FlightDetails/FlightDetailsModal";
 
-const FlightCard: React.FC<{flightResults: FlightResultsProps}> = ({flightResults} : {flightResults: FlightResultsProps}) => {
+type FlightCardProps = {
+    flightResults: FlightResultsProps,
+    showMode?: boolean
+};
+
+/**
+ *
+ * @param flightResults
+ * @param showMode - if true, the component doesn't show the price nor the details modal, if false it shows the
+ * passengers number data.
+ * @constructor
+ */
+const FlightCard: React.FC<FlightCardProps> = ({flightResults, showMode} : FlightCardProps) => {
+    const [open, setOpen] = React.useState(false);
+    const handleOpen = () => setOpen(true);
     let randomString = (Math.random() + 1).toString(36).substring(7);
    return (
-    <Grid data-testid="flightCard" key={randomString + flightResults.id} container item className={flightCardStyles.containerCard} xs={12}>
-        {flightResults.itineraries.map((item) => <FlightInfoComponent key={randomString + item.segments[0].arrival.at + flightResults.id} itineraries={item.segments[0]}/>)}
-        <Grid item xs={12} container flexDirection="row" className={flightCardStyles.amountContainer}>
-            <Grid item xs={12}>
-                <hr className={flightCardStyles.dividerAmount}/>
-            </Grid>
-            <Grid item xs={12} container justifyContent="flex-end">
-                <Grid item xs={4} container justifyContent="flex-end">
-                    <Button variant="outlined" className={flightCardStyles.amountButton}>${flightResults.price.total}</Button>
+       <Grid data-testid="flightCard" key={randomString + flightResults.id} container item className={flightCardStyles.containerCard} xs={12}>
```

Comment:
I suggest to use template literals ``to concatenate a string this gives us more clarity at reading code time :)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals